dbt/adapters/dremio/connections.py

```diff
@@ -231,6 +230,79 @@ def create_catalog(self, relation):
             logger.debug(f"Creating folder(s): {database}.{schema}")
             self._create_folders(database, schema, rest_client)
         return
+    
+    # dbt docs integration with Dremio wikis and tags
+    def docs_integration_with_wikis(self, relation, text: str):
+        logger.debug("Integrating wikis")
+        thread_connection = self.get_thread_connection()
+        connection = self.open(thread_connection)
+        rest_client = connection.handle.get_client()
+        database = relation.database
+        schema = relation.schema
+
+        path = self._create_path_list(database,schema)
+        identifier = relation.identifier
+        path.append(identifier)
+        try:
+            catalog_info = rest_client.get_catalog_item(
+                catalog_id=None,
+                catalog_path=path,
+            )
+        except DremioNotFoundException:
+            logger.debug("Catalog not found. Returning")
+            return
+
+        object_id = catalog_info.get("id")
+        stored_wiki = rest_client.retrieve_wiki(object_id)
+        wiki_content = stored_wiki.get("text")
+        wiki_version = stored_wiki.get("version", None)
+
+        if wiki_version == None:
+            logger.debug(f"Creating wiki for {'.'.join(path)}")
+            logger.debug(rest_client.create_wiki(object_id, text))
+        elif wiki_content != text:
+            if text == "": # text is empty, delete wiki
+                logger.debug(f"Deleting wiki for {'.'.join(path)}")
+                logger.debug(rest_client.delete_wiki(object_id, wiki_version))
+            else:
+                logger.debug(f"Updating wiki for {'.'.join(path)}")
+                logger.debug(rest_client.update_wiki(object_id, text, wiki_version))
+
+    def docs_integration_with_tags(self, relation, tags: Union[str,list[str]]):
+        logger.debug("Integrating tags")
+        thread_connection = self.get_thread_connection()
+        connection = self.open(thread_connection)
+        rest_client = connection.handle.get_client()
+        database = relation.database
+        schema = relation.schema
+
+        path = self._create_path_list(database,schema)
+        identifier = relation.identifier
+        path.append(identifier)
+        try:
+            catalog_info = rest_client.get_catalog_item(
+                catalog_id=None,
+                catalog_path=path,
+            )
+        except DremioNotFoundException:
+            logger.debug("Catalog not found. Returning")
+            return
+
+        object_id = catalog_info.get("id")
+        stored_tags = rest_client.retrieve_tags(object_id)
+        tags_list = stored_tags.get("tags")
+        tags_version = stored_tags.get("version", None)
+
+        if tags_version == None:
+            logger.debug(f"Creating tags for {'.'.join(path)}")
+            logger.debug(rest_client.create_tags(object_id, tags))
```

Comment:
In general have else if is not the best practice. 
Sometimes it's got to have return after one If and there will be no need to have one else. 

The same recommendation may come to the previous method.